### PR TITLE
MRG: Pin required MNE version to >= 0.24, update metadata in setup.cfg

### PR DESCRIPTION
**Pull request by hoechenberger**
Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [x] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -20,22 +20,17 @@ classifiers =
...
-  Programming Language :: Python :: 3.7
-  Programming Language :: Python :: 3.8
-  Programming Language :: Python :: 3.9
 project_urls =
   Documentation = https://mne.tools/mne-bids
   Bug Reports = https://github.com/mne-tools/mne-bids/issues
   Source = https://github.com/mne-tools/mne-bids
 
 [options]
-python_requires = ~= 3.7
```
Comment (sappelhoff) [inline setup.cfg:36]: why remove this?

File: setup.cfg [hunk 2]
```diff
@@ -20,22 +20,17 @@ classifiers =
...
   Topic :: Scientific/Engineering
   Operating System :: Microsoft :: Windows
-  Operating System :: POSIX
-  Operating System :: Unix
+  Operating System :: POSIX :: Linux
   Operating System :: MacOS
   Programming Language :: Python
-  Programming Language :: Python :: 3.7
-  Programming Language :: Python :: 3.8
-  Programming Language :: Python :: 3.9
```
Comment (sappelhoff) [inline setup.cfg:29]: why remove this? It's relatively little effort to update this once per release, that is: add new versions we support, and remove old ones :thinking:

File: setup.cfg (hunk 1 above)
Comment (hoechenberger) [reply setup.cfg:36]: It's implicitly defined through our MNE-Python dependency. I thought we could avoid this redundancy, but can revert if you want to have this

File: setup.cfg (hunk 2 above)
Comment (hoechenberger) [reply setup.cfg:29]: See my response above

File: setup.cfg (hunk 1 above)
Comment (sappelhoff) [reply setup.cfg:36]: I'd like to keep it, please :)

File: setup.cfg (hunk 2 above)
Comment (agramfort) [reply setup.cfg:29]: yes I prefer to be explicit here

**Comment by hoechenberger** [comment]
Ok, I've reverted those classifier changes in `setup.cfg`

**Review by sappelhoff — APPROVED** [review]
thanks!

**Pull request merged**